spec/services/determine_training_record_state_spec.rb

```diff
@@ -2,1504 +2,1644 @@
 
 require "rails_helper"
 
-RSpec.describe DetermineTrainingRecordState, :with_training_record_state_examples do
-  let(:current_school) { nil }
+RSpec.describe DetermineTrainingRecordState, :with_default_schedules do
+  let(:scenarios) { NewSeeds::Scenarios::Participants::TrainingRecordStates.new }
 
-  subject(:determined_state) { described_class.call(participant_profile:, school: current_school) }
+  let!(:current_school) { nil }
 
-  shared_examples "determines states as" do |validation_state, training_eligibility_state, fip_funding_eligibility_state, training_state, record_state|
-    it "sets the correct states" do
+  subject(:determined_state) do
+    TrainingRecordState.refresh
+
+    described_class.call(participant_profile:, school: current_school)
+  end
+
+  shared_examples "determines states as" do |validation_state, training_eligibility_state, fip_funding_eligibility_state, mentoring_state, training_state, record_state|
+    it "states are determined as expected" do
       expect(determined_state.validation_state).to eq validation_state
       expect(determined_state.training_eligibility_state).to eq training_eligibility_state
       expect(determined_state.fip_funding_eligibility_state).to eq fip_funding_eligibility_state
+      expect(determined_state.mentoring_state).to eq mentoring_state
       expect(determined_state.training_state).to eq training_state
       expect(determined_state.record_state).to eq record_state
-      expect(I18n.t("status_tags.admin_participant_status").keys).to include record_state
-      expect(I18n.t("status_tags.appropriate_body_participant_status").keys).to include record_state
-      expect(I18n.t("status_tags.delivery_partner_participant_status").keys).to include record_state
-      expect(I18n.t("status_tags.school_participant_status").keys).to include record_state
-      expect(I18n.t("status_tags.school_participant_status_detailed").keys).to include record_state
+    end
+
+    it "Each StatusTag has a language file entry for the record_state of \"#{record_state}\"" do
```

Comment:
Could we move those expectations back into one `it` block as this will cause the tests to take substantially longer to run?